pkg/controller.v2/trainjob_controller.go

```diff
@@ -36,6 +39,15 @@ import (
 
 var errorUnsupportedRuntime = errors.New("the specified runtime is not supported")
 
+type objsOpState int
+
+const (
+	succeeded      objsOpState = iota
+	buildFailed    objsOpState = iota
+	creationFailed objsOpState = iota
+	updateFailed   objsOpState = iota
```

Comment:
What is the goal to represent conditions as increment int constants ? Is it require less memory than just using strings ?

Reply:
This is Go World ENUM. Shouldn't the state be represented as Enums?
Additionally, the string could be unhandled since the string has loose constraints.
It would be better to manage states as strong constraints.

Reply:
I see, do you know if k8s uses the same representation for the conditions ?

Reply:
> do you know if k8s uses the same representation for the conditions ?

Does this mean that whether or not any K8s controllers (like Deployment Controller) have the same transition calculation mechanism?